contracts/tgrade-valset/src/contract.rs

```diff
@@ -532,6 +538,19 @@ fn end_block(deps: DepsMut, env: Env) -> Result<Response, ContractError> {
     // calculate and store new validator set
     let (validators, auto_unjail) = calculate_validators(deps.as_ref(), &env)?;
 
+    // update operators list with info about whether or not they're active validators
+    let ops: Vec<_> = operators()
+        .keys(deps.storage, None, None, Order::Ascending)
+        .collect::<StdResult<_>>()?;
+    for op in ops {
+        let active = validators.iter().any(|val| val.operator == op);
```

Comment:
Not very efficient, but since the validators list is expected to be small, it probably doesn't matter?

Reply:
Actually, if you'll look at my PR I'm fighting similar inefficiency issue: https://github.com/confio/poe-contracts/pull/35#discussion_r791841093

Validators amount may be up to 300, so it doesn't seem much - but this is end block. So every end block there will be loop within loop for X(<=300) validators, and in worst case there will be 3 additional loops from me. :)
I think this starts to add up.

Not necessarily disarding your implementation, just pointing that out.

I still think that all those separate loops (along with calculating validators diff at 562) could be made into more tight implementation - although probably much more complex..

Reply:
Iterating over list is fine.
Querying state an extra time for each validator would be a bit more troublesome